### ISSUE TITLE
Add test coverage for tests/helpers.py

### ISSUE THREAD

**Issue by abn** [python-poetry/poetry#9161]
As requested by @Secrus in https://github.com/python-poetry/poetry/pull/9149#discussion_r1525588514.

**Comment by dimbleby** [comment]
I don't understand.  If there is code in tests/helpers.py that is not executed, just remove it already?

**Comment by DarkTechPirate** [comment]
No what are even supposed to do ?

**Comment by abn** [comment]
@dimbleby I think @Secrus is asking for test helpers to have functional test coverage. But yes, if we are not using a helper we should remove it. Although, I do not think that is the case right now.

**Comment by dimbleby** [comment]
I dont buy it: who writes tests for their tests?  what then, shall we write tests for the tests for the tests too?

test code automatically gets coverage by being executed in the test suite, else delete it

**Comment by abn** [comment]
I'll let @Secrus decide what they want to do with this one as I just created the issue based on his request. :wink:

**Comment by Julynx** [comment]
> I dont buy it: who writes tests for their tests? what then, shall we write tests for the tests for the tests too?
> 
> test code automatically gets coverage by being executed in the test suite, else delete it

This is a horrible take. Tests should NEVER dictate which code is in your application by their coverage.

Imagine you have code that looks like this:

```python
if len(text) > MAX_LEN:
    raise ValueError("Text too long.")

do_something()
```

And you're getting coverage 90% in your tests because they are crap and are only sending text shorter than MAX_LEN, so your if branch is never triggered.

So, by your logic we should go "lets remove the code that isnt covered then". 

```python
do_something()
```

Boom, 100% coverage.

And now a user will send text longer than MAX_LEN, and your app will crash because it wasn't expecting it.

Coverage is meant for you to implement all possible cases and then ensure you test all of them in your tests. Its not something you can solve by removing those cases and hoping they never happen in production. 

TLDR. To get coverage to 100% you make decent tests instead of removing random important lines of code from your application.

**Comment by dimbleby** [comment]
@Julynx you have completely misunderstood what is being discussed here

**Comment by Julynx** [comment]
> @Julynx you have completely misunderstood what is being discussed here

Fair, I apologize for the misunderstading and I'm deleting my previous reply. Thanks for your patience.

**Comment by j7an** [comment]
@python-poetry/triage main branch has merged `test_helpers.py`. This issue should be closed as resolved.
